src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -811,52 +811,58 @@ DeferredErrorIfUnsupportedRecurringTuplesJoin(
 				 */
 				recurType = FetchFirstRecurType(plannerInfo, outerrelRelids);
 
-				break;
+				if (recurType == RECURRING_TUPLES_REFERENCE_TABLE)
```

Comment:
Regarding `DeferredErrorIfUnsupportedRecurringTuplesJoin `, let me share our internal discussions for visibility here.

Outer join checks in `DeferredErrorIfUnsupportedRecurringTuplesJoin ` actually becomes obsolete, as we are now able to support those.

However, keeping them for now might make sense for LATERAL join checks.

For example, due to #5753, we should be very careful about queries like: 
```SQL
select 
    count(*) 
    from 
    ref 
   LEFT JOIN LATERAL 
    (SELECT sum(i) FROM dist WHERE dist.i = ref.i GROUP BY dist.j) as foo ON (True); 
``` 

We should throw error for these cases. And, the current checks in `DeferredErrorIfUnsupportedRecurringTuplesJoin` could actually find the outer joins that could not be recursively planned. 

Though, that might be little confusing for the readers. Best is to add tight error checks as we do in #5753

Reply:
done [e42b8ab](https://github.com/citusdata/citus/pull/6512/commits/e42b8ab9f47656b4d0b36329ad02e9b5dd6f9c86)